replicated_kv_store/client.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"github.com/krithikvaidya/distributed-dns/replicated_kv_store/protos"
+)
+
+//WriteCommand allows clients to submit a new command to the leader
+func (node *RaftNode) WriteCommand(operation []string) bool {
+	node.raft_node_mutex.Lock()
+	defer node.raft_node_mutex.Unlock()
+
+	// True only if leader
+	if node.state == Leader {
+		//append to local log
+		node.log = append(node.log, protos.LogEntry{Term: node.currentTerm, Operation: operation})
+
+		var entries []*protos.LogEntry
+		entries = append(entries, &node.log[len(node.log)-1])
+
+		msg := &protos.AppendEntriesMessage{
+
+			Term:         node.currentTerm,
+			LeaderId:     node.replica_id,
+			PrevLogIndex: int32(len(node.log) - 1),
+			PrevLogTerm:  node.log[len(node.log)-1].Term,
+			LeaderCommit: node.commitIndex,
+			Entries:      entries,
+		}
+
+		node.LeaderSendAEs(operation[0], msg, int32(len(node.log)-1))
```

Comment:
LeaderSendAEs will perform the sending of the AppendEntries asynchronously (uses separate goroutines to send the AE to each replica), and will return before the AppendEntries is completed. We need to ensure that a majority (including the leader) have successfully responded to the write before returning true. I guess you can use another channel for this purpose, and make some changes to LeaderSendAEs.